src/libostree/ostree-mutable-tree.c

```diff
@@ -299,6 +446,74 @@ ostree_mutable_tree_ensure_parent_dirs (OstreeMutableTree  *self,
   return ret;
 }
 
+const char empty_tree_csum[] = "6e340b9cffb37a989ca544e6bb780a2c78901d3fb33738768511a30617afa01d";
+
+static void
+set_string(gchar **out, const gchar *in)
+{
+  if (g_strcmp0(*out, in) == 0)
+    return;
+  g_free (*out);
+  *out = g_strdup (in);
+}
+
+/**
+ * ostree_mutable_tree_fill_empty_from_dirtree:
```

Comment:
Possibly a better name would be `ostree_mutable_tree_fast_merge`?  It's more intention revealing.  Or perhaps this should be moved into a `-private.h` header and not be made part of the public API?

Reply:
Yeah, I would agree with that (keeping it private for now).